stores/ProgressStore.js

```diff
@@ -0,0 +1,82 @@
+import { defineStore } from "pinia";
+
+export const useProgressStore = defineStore("progress", {
+  state: () => ({
+    progress: 0,
+    currentLevel: 1,
+    levelCount: 2,
+    initialized: false,
+  }),
+  getters: {
+    currLevelString(state) {
+      return `${state.currentLevel}`;
+    },
+    accessGranted(state) {
+      return (newLevel) => accessCheck(state.progress, newLevel);
+    },
+    // Only returns the next level string if it is accessible
+    nextLevelString(state) {
+      return `${
+        accessCheck(state.progress, state.currentLevel + 1)
+          ? state.currentLevel + 1
+          : state.currentLevel
+      }`;
+    },
+    prevLevelString(state) {
+      return `${
+        accessCheck(state.progress, state.currentLevel - 1)
+          ? state.currentLevel - 1
+          : state.currentLevel
+      }`;
+    },
+  },
+  actions: {
+    init() {
+      // Necessary for SSR
+      if (typeof window !== "undefined") {
+        this.progress = parseInt(localStorage.getItem("progress")) || 0;
+        this.currentLevel = parseInt(useRoute().fullPath.split("/")[2]) || 1;
+        this.initialized = true;
+      }
+    },
+    levelUp() {
+      this.progress = Math.max(this.progress, this.currentLevel);
+      localStorage.setItem("progress", this.progress);
+    },
+    // Does nothing as long as the level is inaccessable
```

Comment:
Wenn du schon Kommentare schreibst, dann warum nicht als JSDoc :) (ebenso ein paar Zeilen darunter)
```suggestion
    /** Does nothing as long as the level is inaccessable */
```